motionplan/motionPlanner.go

```diff
@@ -86,24 +86,54 @@ func (mp *planner) checkPath(planOpts *PlannerOptions, seedInputs, target []fram
 	return ok
 }
 
-// needed to wrap slices so we can use them as map keys.
-type node struct {
+// node interface is used to wrap a configuration for planning purposes.
+type node interface {
+	// return the configuration associated with the node
+	Q() []frame.Input
```

Comment:
Linter is likely to complain about public member of non-public interface

Reply:
It didn't complain about this... should I fix anyway?

Reply:
Never mind- the linter rule is for public members of non-public structs, not interfaces. Which makes sense. Disregard.